File: est datos/cpp/insercion.cpp
#include <iostream>
using namespace std;
int main(int argc, char const *argv[])
{
    int arr[] = {5,8,2,6,1};

    int pos = 3;

    int dato = 10;

    for (int i = 5; i>= pos; i--){
        arr[i] = arr[i-1];
    }

    arr[pos-1] = dato;

    for (int i=0 ; i<sizeof(arr)/sizeof(arr[0]);i++){
        cout << arr[i] << endl;
    }



    return 0;
}

File: est datos/c#/insercion.cs
    using System;
using System.ComponentModel.Design;
class Program{
    static void Main(string[] args)
    {
        int[] arr = [123,31,2,21,423,56];
        
        Console.WriteLine("Se tiene el arreglo: " + string.Join(",",arr));

        Console.WriteLine("Ingrese el dato que desea insertar:");
        int dato = Convert.ToInt32(Console.ReadLine());
        Console.WriteLine("Ingrese la posicion en la que desea insertar el dato:");
        int pos = Convert.ToInt32(Console.ReadLine());

        for (int i = arr.Length - 1; i >= pos; i--) {
            arr[i] = arr[i - 1];
        }
        arr[pos-1] = dato;

        Console.WriteLine("El nuevo arreglo es: \n" + string.Join(",",arr));



    }
}